hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieNativeAvroHFileReader.java

```diff
@@ -338,10 +339,24 @@ private HFileReader newHFileReader() throws IOException {
     return new HFileReaderImpl(inputStream, fileSize);
   }
 
+  private Option<BloomFilter> getBloomFilter() {
+    Option<BloomFilter> bloomFilter = Option.empty();
+    if (config.enabled()) {
```

Comment:
for an existing table, there are chances older files may not have the bloom filter, but only the new ones will have right. So, we can't rely on write properties

Reply:
Bloom filters are being written by default for hudi tables. I dont see any control flags for writing bloom filter in HoodieAvroFileWriterFactory.

Also, tested with a current pipeline that did not have this code and config enabled before.

Reply:
Added a flag to control whether to use bloom filter for full-key lookup.